tortoise/contrib/pydantic/creator.py

```diff
@@ -517,3 +350,397 @@ def pydantic_queryset_creator(
+                            (
+                                    json_schema_extra.get("nullable")
+                                    and not is_to_one_relation
+                                    and field_description.field_type not in (IntField, TextField)
+                            )
+                            or (self._exclude_read_only and json_schema_extra.get("readOnly"))
+                    ):
+                        fconfig["default_factory"] = lambda: None
+                    self._properties[field_name] = (ftype, Field(**fconfig))
+
+    def _process_normal_field_description(
+            self,
+            field_name: str,
+            field_description: FieldDescriptionBase,
+            json_schema_extra: Dict[str, Any],
+            fconfig: Dict[str, Any],
+    ) -> Tuple[Optional[Any], bool]:
+        if isinstance(field_description, (BackwardFKRelationDescription, ManyToManyFieldInstanceDescription)):
+            return self._process_many_field_relation(field_name, field_description), False
+        elif isinstance(
+                field_description,
+                (
+                        ForeignKeyFieldInstanceDescription,
+                        OneToOneFieldInstanceDescription,
+                        BackwardOneToOneRelationDescription
+                )
+        ):
+            return self._process_single_field_relation(field_name, field_description, json_schema_extra), True
+        elif field_description.field_type is JSONField:
+            return self._process_json_field_description(), False
+        return self._process_data_field_description(field_name, field_description, json_schema_extra, fconfig), False
+
+    def _process_single_field_relation(
+            self,
+            field_name: str,
+            field_description: Union[
+                ForeignKeyFieldInstanceDescription,
+                OneToOneFieldInstanceDescription,
+                BackwardOneToOneRelationDescription
+            ],
+            json_schema_extra: Dict[str, Any],
+    ) -> Optional[Type[PydanticModel]]:
+        model: Optional[Type[PydanticModel]] = self._get_submodel(field_description.python_type, field_name)
+        if model:
+            if field_description.nullable:
+                json_schema_extra["nullable"] = True
+            if field_description.nullable or field_description.default is not None:
+                model = Optional[model]  # type: ignore
+
+            return model
+        return None
+
+    def _process_many_field_relation(
+            self,
+            field_name: str,
+            field_description: Union[BackwardFKRelationDescription, ManyToManyFieldInstanceDescription],
+    ) -> Optional[Type[List[Type[PydanticModel]]]]:
+        model = self._get_submodel(field_description.python_type, field_name)
+        if model:
+            return List[model]  # type: ignore
+        return None
+
+    def _process_json_field_description(self) -> Any:
+        return Any
```

Comment:
this is the same as it is right now, also claimed here: #1702

I made it to an actual function so that in the future someone could add the functionality described there.

Reply:
removed this method